enhancements/insights/container-scanner.md

```diff
@@ -0,0 +1,498 @@
+
+At most, the additional data adds __~200 uncompressed bytes per WorkloadContainerShape in the JSON payload__.
+The current maximum number of pods that are checked by the Insights Operator is 8000 pods.
+
+This additional payload does not scale with the number of pods but with the number of `WorkloadContainerShape` which can be approximated as the number of applications (deployments,statefulsets,...) times the number of containers per pod.
+
+#### UX Risk & Mitigation
+
+The additional runtime info added to the gathered data will be exposed to the user though the archives generated by the Insights operator. As the values of the field are hashed, the user will not be able to determine their corresponding content.
+
+Exposing these data to the user with a clean user interface (eg within console.redhat.com) is out of scope of this proposal.
+
+### Drawbacks
+
+If implemented, Red Hat will document the addition of this new collection. We  introduce the risk that some number of customers who would have or are sending telemetry data via the Insights operator will choose to turn it off because they are uncomfortable sharing this additional data. 
+
+The container scanner needs to be continually updated to detect new workloads  that Red Hat wants information about. 
+This can be done iteratively and will require an updated reference to the container scanner image in the Insights Operator to take into account the additional detections.
+
+As the container scanner is sanitizing the data extracted from the containers, it is not able to retrieve "raw" data (such as processes' name and command line) which limits its ability to detect emerging trends. The container scanner must provide fingerprints executables for workloads that Red Hat wants to monitor.
+
+## Design Details
+
+### Open Questions [optional]
+
+#### Container Scanner Image Availability
+
+The Insights Operator must be able to deploy the container scanner from an image that is available by default in the cluster image registry.
+
+How can we configure a _related image_ in the Insights Operator from its `ClusterVersion` What is the equivalent of a ClusterServiceVersion (CSV)’s `.spec.relatedImages` for OpenShift cluster operators?
+
+## Test Plan
+
+### Container Scanner Test Plan
+
+The container scanner can be tested in isolation to verify its capability to scan containers and collect the expected runtime info.
+
+Testing the container scanner can be done outside of the Insights Operator test suite.
+It requires integration tests with applications based on the detectable runtime stacks.
+
+### Insights Operator Test Plan
+
+The integration tests for the Insights Operator will verify that it is able to properly deploy the container scanner as a `DaemonSet` and have some smoke tests to validate that the additional runtime info can effectively be collected.
+
+## Graduation Criteria
+
+TBD
+
+### Dev Preview -> Tech Preview
+
+* Make the container scanner disabled by default
```

Comment:
If this will be behind the featuregate then this doesn't make much sense to me.

Reply:
@tremes thanks, I've updated the doc